src/main/java/com/Crowdsourcing/port_tasks/CrowdsourcingPortTasks.java

```diff
@@ -0,0 +1,184 @@
+		VarbitID.PORT_TASK_SLOT_4_ID, 4
+	);
+
+	@SuppressWarnings("unused")
+	@Subscribe
+	private void onMenuOptionClicked(final MenuOptionClicked event)
+	{
+		final int identifier = event.getId();
+		// If jagex adds more notice boards we'll send null
+		// But for now it sends the notice board the player is at
+		currentNoticeboard = null;
+		if (NOTICEBOARDS.contains(identifier))
+		{
+			currentNoticeboard = identifier;
+		}
+	}
+
+	@SuppressWarnings("unused")
+	@Subscribe
+	private void onVarbitChanged(final VarbitChanged event)
+	{
+		final int varbitId = event.getVarbitId();
+		final int value = event.getValue();
+		final Integer idx = TASK_SLOT_INDEX.get(varbitId);
+		if (idx == null)
+		{
+			return; // Not a port task assignment
+		}
+
+		currentTasks[idx] = value;
+		// A value of 0 means our task was completed and slot is empty
+		// Only add actual assignments to the queue
+		if (value > 0)
+		{
+			addTask(value);
+		}
+	}
+
+	@SuppressWarnings("unused")
+	@Subscribe
+	private void onWidgetLoaded(final WidgetLoaded event)
+	{
+		if (event.getGroupId() != InterfaceID.PORT_TASK_BOARD)
+		{
+			return;
+		}
+		// Interface isn't ready on widget load and needs to be done later
+		clientThread.invokeLater(this::scanPortTaskBoard);
+	}
+
+	private void scanPortTaskBoard()
+	{
+		final Widget widget = client.getWidget(InterfaceID.PortTaskBoard.CONTAINER);
+		if (widget == null)
+		{
+			return;
+		}
+		List<Widget> children = new ArrayList<>();
+		if (widget.getDynamicChildren() != null)
+		{
+			children.addAll(Arrays.asList(widget.getDynamicChildren()));
+		}
+
+		lastOfferedTasks.clear();
+		lastOfferedTasks.addAll(offeredTasks);
+		offeredTasks.clear();
+
+		for (Widget child : children)
+		{
+			// The notice board is made up of 48 dynamic children
+			if (child == null)
+			{
+				continue;
+			}
+			// The clickable tasks are the only widgets with opListeners
+			Object[] ops = child.getOnOpListener();
+			if (ops == null || ops.length < 4)
+			{
+				continue;
+			}
+			// The value of the last opListener is the dbrow of the offered task
+			offeredTasks.add((Integer) ops[3]);
+		}
+		if (!offeredTasks.equals(lastOfferedTasks))
+		{
+			submitTasks();
+		}
+	}
+
+	private void addTask(int value)
+	{
+		if (lastTasks.size() == 20)
+		{
+			lastTasks.removeFirst();
+		}
+		lastTasks.addLast(value);
+	}
+
+	private void submitTasks()
+	{
+		if (client == null || client.getLocalPlayer() == null)
+		{
+			return;
+		}
+		// No need to use boat location as you can't interact with notice boards in a boat or instance
+		LocalPoint local = LocalPoint.fromWorld(client, client.getLocalPlayer().getWorldLocation());
+		WorldPoint location = WorldPoint.fromLocal(client, local);
+
+		PortTaskData data = new PortTaskData(
+			location,
+			client.getRealSkillLevel(Skill.SAILING),
+			currentNoticeboard,
+			Arrays.copyOf(currentTasks, currentTasks.length),
+			List.copyOf(offeredTasks),
+			new ArrayDeque<>(lastTasks)
```

Comment:
Just make this a List in the data obj. I have no clue how array deque is serialized offhand and afaict it isn't necessary (I'd assume just like an array)